### PR TITLE
Update the styling of the Numeric Input component to match other Source input components

### PR DESCRIPTION
**Pull request by i-hardy**
## What are you changing?

- Updating the border styling and style of various input states on the `NumericInput` component

## Why?

- The various input components in base Source have been updated to a new border design, and this component needs to match the new look.

### CODE REVIEW

File: libs/@guardian/source-react-components-development-kitchen/src/numeric-input/inputExtensionStyles.ts [hunk 1]
```diff
@@ -1,20 +1,26 @@
+import type { SerializedStyles } from '@emotion/react';
 import { css } from '@emotion/react';
 import { space } from '@guardian/source-foundations';
 import { textInputThemeDefault } from '@guardian/source-react-components';
+import type { InputTheme } from './NumericInput';
 import { inputBase } from './sharedStyles';
 
-export const inputExtension = (input = textInputThemeDefault.textInput) => css`
+export const inputExtension: (
+	input: InputTheme['textInput'],
```
Comment (jamesmockett) [inline libs/@guardian/source-react-components-development-kitchen/src/numeric-input/inputExtensionStyles.ts:9]: Could you do the type annotation this way instead?

```
export const inputExtension = (input = textInputThemeDefault.textInput): SerializedStyles => css`
```

**Review by jamesmockett — COMMENTED** [review]
Thanks Imogen, this looks great 🙏  Before merging I think we should get @akemitakagi to take a look and sign off the visual changes.

**Comment by akemitakagi** [comment]
Hello! They look great. I reviewed the Chromatic build and accepted it.

**Review by joecowton1 — APPROVED** [review]

**Comment by joecowton1** [comment]
Nice work, thanks @i-hardy 🙏

**Pull request merged**